tendermint/src/lite_impl/header.rs

```diff
@@ -48,28 +48,80 @@ impl lite::Header for block::Header {
         fields_bytes.push(encode_hash(&self.validators_hash));
         fields_bytes.push(encode_hash(&self.next_validators_hash));
         fields_bytes.push(encode_hash(&self.consensus_hash));
-        fields_bytes.push(bytes_enc(&self.app_hash));
+        fields_bytes.push(encode_bytes(&self.app_hash));
         fields_bytes.push(self.last_results_hash.as_ref().map_or(vec![], encode_hash));
         fields_bytes.push(self.evidence_hash.as_ref().map_or(vec![], encode_hash));
-        fields_bytes.push(bytes_enc(self.proposer_address.as_bytes()));
+        fields_bytes.push(encode_bytes(self.proposer_address.as_bytes()));
 
         Hash::Sha256(simple_hash_from_byte_vectors(fields_bytes))
     }
 }
 
-fn bytes_enc(bytes: &[u8]) -> Vec<u8> {
-    let mut chain_id_enc = vec![];
-    prost_amino::encode_length_delimiter(bytes.len(), &mut chain_id_enc).unwrap();
```

Comment:
The problem was here: we length-encoded even zero length bytes. In this particular case the `app_hash`.